main.js

```diff
@@ -50,52 +74,83 @@ async function main() {
   await exec.exec("make install", undefined, {
     cwd: luaRocksExtractPath
   })
+}
+
+async function main() {
+  const luaRocksVersion = core.getInput('luaRocksVersion', { required: true })
+
+  const luaRocksInstallPath = path.join(process.cwd(), LUAROCKS_PREFIX)
+
+  const tempBuildPath = path.join(process.env["RUNNER_TEMP"], BUILD_PREFIX)
+  await io.mkdirP(tempBuildPath)
+
+  let luaPath = core.getInput("withLuaPath")
+  if (!luaPath) {
+    // NOTE: this is the default install path provided by gh-actions-lua
+    luaPath = path.join(process.cwd(), LUA_PREFIX)
+  }
+
+  core.addPath(path.join(luaRocksInstallPath, "bin"));
+
+  if (isWindows()) {
+    await installWindows(luaRocksVersion, tempBuildPath, luaRocksInstallPath, luaPath);
+  } else {
+    await installUnix(luaRocksVersion, tempBuildPath, luaRocksInstallPath, luaPath);
+  }
 
   // Update environment to use luarocks directly
-  let lrPath = ""
+  let lrBin = ""
 
-  await exec.exec(`${path.join(luaRocksInstallPath, "bin", "luarocks")} path --lr-bin`, undefined, {
+  await exec.exec("luarocks path --lr-bin", undefined, {
     listeners: {
       stdout: (data) => {
-        lrPath += data.toString()
+        lrBin += data.toString()
       }
     }
   })
 
-  if (lrPath != "") {
-    core.addPath(lrPath.trim());
+  await exec.exec("luarocks path --lr-bin", undefined, {
+    listeners: {
+      stdout: (data) => {
+        lrBin += data.toString()
+      }
+    }
+  })
+
+  if (lrBin != "") {
+    core.addPath(lrBin.trim());
   }
 
-  let luaPath = ""
+  let lrPath = ""
 
   await exec.exec("luarocks path --lr-path", undefined, {
     listeners: {
       stdout: (data) => {
-        luaPath += data.toString()
+        lrPath += data.toString()
       }
     }
   })
 
-  luaPath = luaPath.trim()
+  lrPath = lrPath.trim()
 
-  let luaCpath = ""
+  let lrCpath = ""
 
   await exec.exec("luarocks path --lr-cpath", undefined, {
     listeners: {
       stdout: (data) => {
-        luaCpath += data.toString()
+        lrCpath += data.toString()
       }
     }
   })
 
-  luaCpath = luaCpath.trim()
+  lrCpath = lrCpath.trim()
 
-  if (luaPath != "") {
-    core.exportVariable("LUA_PATH", ";;" + luaPath)
+  if (lrPath != "") {
+    core.exportVariable("LUA_PATH", ";;" + lrPath)
   }
 
-  if (luaCpath != "") {
-    core.exportVariable("LUA_CPATH", ";;" + luaCpath)
+  if (lrCpath != "") {
+    core.exportVariable("LUA_CPATH", ";;" + lrCpath)
   }
```

Comment:
I've been using this workflow, and this seems to generate a unix style path (with semicolons as separators, instead of colons).

```
> printenv
LUA_PATH=;;C:\Users\runneradmin\AppData\Roaming/luarocks/share/lua/5.1/?.lua;C:\Users\runneradmin\AppData\Roaming/luarocks/share/lua/5.1/?/init.lua

PATH=/c/Program Files/PowerShell/7:/c/Users/runneradmin/AppData/Roaming/luarocks/bin:/d/a/rocks-binaries/rocks-binaries/.luarocks/bin:/d/a/rocks-binaries/rocks-binaries/.lua/bin:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/Extensions/Microsoft/IntelliCode/CLI:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/VC/VCPackages:/c/Program Files (x86)/Microsoft Visual 
...
```

As a result, build-time dependencies can't be found (e.g. when installing `toml-edit`, which depends on `luarocks-build-rust-mlua`.

Reply:
@mrcjkb  did you solve this issue?

Reply:
No, not yet.

Reply:
@mrcjkb LUA_PATH uses `;` on both Windows and Unix, and PATH uses `:` on Unix and `;` on Windows. I don't think the semicolons are the cause of your issue, but [there were issues related to build dependency lookup](https://github.com/luarocks/luarocks/commit/ea43975308abc8b5281ac8934657aa4d59bcdd75) that were fixed in 3.10.0.

Reply:
🙏 I can confirm it works when I add

```yaml
with:
    luarocksVersion: "3.10.0"
```

Reply:
Actually, no it doesn't. I was looking at the Linux builder.
The Windows builder still fails when trying to use the `luarocks-build-rust-mlua` build backend: 

```
Error: Failed initializing build back-end for build type 'rust-mlua': module 'luarocks.build.rust-mlua' not found:
	No LuaRocks module found for luarocks.build.rust-mlua
```

https://github.com/nvim-neorocks/rocks-binaries/actions/runs/8168782585/job/22331528480#step:11:1256

Reply:
@mrcjkb Please try again bumping the version number to `3.11.0`! I've uploaded a pre-release build but I haven't announced it yet. :shushing_face: :wink:

Reply:
That has fixed it! :pray: :pray: :pray: 
:tada: :rocket: